inference-engine/src/vpu/graph_transformer/src/middleend/passes/merge_permute_stages.cpp

```diff
@@ -55,11 +55,24 @@ class PassImpl : public Pass {
                                   permuteStage->output(0)->desc().dimsOrder());
     }
 
-    static bool isTrivialPermute(const PermutationIndexVector& permuteDims) {
-        for (size_t i = 0; i < permuteDims.size(); ++i)
-            if (i != permuteDims[i])
-                return false;
+    static bool isTrivialPermute(const PermutationIndexVector& permuteDims, const vpu::DimValues& dims) {
+        const auto defDims = DimsOrder::fromNumDims(dims.size());
+        InferenceEngine::SizeVector dimsVector(dims.size());
+        for (const auto& dim : dims) {
+            dimsVector[defDims.dimInd(dim.first)] = dim.second;
```

Comment:
Have you tried `dimToIeInd` function for this?

Reply:
Done